platform/core.windows/src/org/netbeans/core/windows/view/dnd/DragWindow.java

```diff
@@ -44,6 +46,9 @@
  * @author sa
  */
 class DragWindow extends JWindow {
+    /* Store buffers at 2x the logical resolution. Then scale them down by 50% when painting to the
+    JWindow. This ensures full-resolution painting on HiDPI and Retina screens. */
+    private static final int DPI_SCALE = 2;
 
     private boolean useFadeEffects = !Boolean.getBoolean( "winsys.dnd.nofadeeffects" );
```

Comment:
the transparent drag window felt always a little slow to me. I think part of it is the fairly long fade-in effect (2s?) which makes it look less snappy as it could be. (easier to see with dark theme).

Wondering if we should turn it off by default or reduce the fade-in time?

(The second problem turned out to be my system settings which also have a small effect for opening/closing windows which clashed with this usecase)

Reply:
Hmm, I'm not seeing much of what the fade effect is doing, on either Windows or MacOS, despite it being confirmed to be on. In general I tend to prefer to remove all animations that don't have a very specific purpose. The drag image itself is enough to show what is going on. Should I just remove the animation?

Reply:
maybe it was added to workaround flickering? It seems to flash white for a split second before it paints the thumbnail. Easy to spot when a dark theme is enabled and animation is off.

Reply:
flickering is fixed by setting the background color of the window. Tested with `setBackground(Color.RED);` which caused it to flicker red.

`JWindow` might not have look and feel color assigned?

Reply:
Ah, now I see what the fade is doing, and I see the flickering you mention too on dark mode. But it's easy to remedy... it's just a matter of setting the window background color appropriately for the light vs. dark LAF.

I will try to make a version that rips out the complicated animation code while avoiding the flickering.

Reply:
while reading through the code i noticed that window transparency is implemented via JNA (DragAndDropFeedbackVisualizer). This could probably use Java 7 API but this is a bit out of scope for this PR i think.

Reply:
I added two more proposed commits, (1) to remove the animation, and (2) to remove the flicker and one more bug I discovered.

We could try to remove the JNA stuff, but that requires testing on every platform, so I didn't do it currently.